client/pages/route-details.jsx

```diff
@@ -181,16 +181,18 @@ export default class RouteDetails extends React.Component {
             </div>
           </div>
           <div className="options-col">
-            <a className="option-link" href="#share-form">
+            <a className="option-link" href="#share-route">
               <div className="option-button">
                 <i className="share-icon fas fa-share" />
                 <span className="button-text">Share</span>
               </div>
             </a>
-            <div className="option-button">
-              <i className="edit-icon fas fa-edit" />
-              <span className="button-text">Edit</span>
-            </div>
+            <a className="option-link" href="#edit-route">
```

Comment:
I noticed that you have a few `setRouteId` method in App, but when you are doing hash routing, you should be able to derive the id of a specific item from the `route.params`.

```jsx
<a className="option-link" href={`#edit-route?${this.state.route.id}`}>
```

Take a close look at how `react-hash-routing` works, the user should be able to refresh the screen and not lose their place in the app. For example going to a specific route should switch the used to `#route-details?routeId=something`. Each component should fetch the data it needs based on the params in the hash. 

At present, you have a large amount of your application's state in `App`, but take a look at `react-hash-routing` and `react-context`. The app only knows 2 things:

- who the user is (if there are separate user accounts)
- what page to show

Everything else should be handled by each screen independently.